LittleHelpBookData/little_help_book.py

```diff
@@ -0,0 +1,106 @@
+import os
+import sys
+from datetime import datetime
+import re
+import numpy as np
+import requests
+from bs4 import BeautifulSoup
+import pandas as pd
+from pymongo import MongoClient, errors
+from tqdm import tqdm
+
+_i = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+if _i not in sys.path:
+    # add parent directory to sys.path so utils module is accessible
+    sys.path.insert(0, _i)
+del _i  # clean up global name space
+from shared_code.utils import (
+    check_similarity, locate_potential_duplicate,
+    insert_services, client
+)
+from shared_code.base_scraper import BaseScraper
+
+
+class LHBScraper(BaseScraper):
+
+    def scrape_updated_date(self):
+
+        resp = super().scrape_updated_date()
+        soup = BeautifulSoup(resp, 'html.parser')
+        date_string = soup.find('relative-time', class_='no-wrap')
+        date_string = date_string.attrs['datetime']
+        date_string = re.search(
+            r'(19|20)\d\d[- /.](0[1-9]|1[012])[- /.](0[1-9]|[12][0-9]|3[01])',
+            date_string
+        ).group(0)
+        scraped_update_date = datetime.strptime(
+            date_string, '%Y-%m-%d'
+        )
+        return scraped_update_date.date()
+
+
+    def grab_data(self):
+
+        '''Subcategory would be serviceSummary, Servicename would be name, PhoneNumber would be phone,
+        Physical Address should be split into address1, city, state, zip, Hours of operation is schedule,
+         Description can be description, Web Address would be website, Email Address would be contact Email'''
+
+
+        df = pd.read_csv(self.data_url, usecols=self.extract_usecols)
+        df.drop_duplicates(
+            subset=self.drop_duplicates_columns,
+            inplace=True,
+            ignore_index=True
+        )
+        df.rename(columns=self.rename_columns, inplace=True)
+        df['phone'] = df['phone'].str.replace(r'(\d{3})\s*[-.]\s*(\d{3}\s*[-.]\s*\d{4})', r'(\1) \2')
+        df.replace('', np.nan, inplace=True)
+        df.dropna(subset=['name'], inplace=True)
+        df[['address1', 'state', 'zip']] = df['Physical Address'].str.extract(r'(.+)(OR).+(\d{5})', expand=True)
+        df['serviceSummary'] = np.where(df['serviceSummary'].isnull(),
+                                        df['Category'],
+                                        df['serviceSummary'])
+        df.drop(['Physical Address', 'Category'], axis=1, inplace=True)
+        df = df.groupby(['name', 'phone'], as_index=False).agg({'serviceSummary': ', '.join, 'address1': 'first',
```

Comment:
Nice, like it. Can we move columns which into groupby , instead of first, will it be faster?? e.g. `
df.groupby(['name', 'address1', 'phone', 'city', 'state', 'zip'....]).agg({'serviceSummary': ', '.join})`

Reply:
Also such a functionality is needed for every scraper , that we want to aggregate on service_summary , how can we extend such a functionality available for every scraper? May be separate this logic into to aggregate_service_summary() method() which takes a list of group by columns. Then we can call that method in the end of main_scraper before inserting into mongodb.

Reply:
The reason why I didn't do groupby in every column as you suggest is because some data might have different description, or missing some other field, which would make it give a wrong result. Choosing only data that is never missing as phone and name we make sure that they are grouped correctly. But this is just my understanding, if I am doing it wrong please tell me! Thanks :)